specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json

```diff
@@ -392,6 +417,45 @@
     }
   },
   "definitions": {
+    "DedicatedHsmOperation": {
```

Comment:
isDataAction is required, please see: https://github.com/Azure/azure-resource-manager-rpc/blob/fbb33d93f0b2cc58ef345c3a430766632ad23936/v1.0/proxy-api-reference.md#exposing-available-operations

Reply:
Looks like it will go to control plane because it's a Resource Provider operation, so I set isDataAction to false

Reply:
@KrisBash 
I'm getting an error about using isDataAction
https://dev.azure.com/azure-sdk/public/_build/results?buildId=356278&view=logs&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d&t=923b5a90-e462-5994-c166-86509c6bf5f2&l=56

Reply:
I see the issue. `isDataAction` is not a valid property of the definition, which is why autorest is failing. It is a property in the operationsAPI schema. It should look like this: 
```
  "definitions": {
    "DedicatedHsmOperation": {
      "description": "REST API operation",
      "isDataAction": false,
      "type": "object",
      "properties": {
        "name": {
          "description": "The name of the Dedicated HSM Resource Provider Operation.",
          "type": "string"
        },
        "isDataAction": {
          "description": "Gets or sets a value indicating whether it is a data plane action",
          "type": "string",
          "readOnly": true
        },
        "display": {
          "properties": {
            "provider": {
              "description": "The Resource Provider of the operation",
              "type": "string"
            },
            "resource": {
              "description": "Resource on which the operation is performed.",
              "type": "string"
            },
            "operation": {
              "description": "Operation type: Read, write, delete, etc.",
              "type": "string"
            },
            "description": {
              "description": "The object that represents the operation.",
              "type": "string"
            }
          }
        }
      }
    },
```

[Here's an example](https://github.com/Azure/azure-rest-api-specs/blob/1bd750a9af598e401a3a3c1c33cd137bfb8cb225/specification/customerlockbox/resource-manager/Microsoft.CustomerLockbox/preview/2018-02-28-preview/customerlockbox.json#L208)

Reply:
Thanks, @KrisBash , sending another PR